dashboard/src/model/Device.php

```diff
@@ -0,0 +1,167 @@
+{
+    private $batteryLevel;
+    private $batterySecondsLeft;
+    private $lastMeasurement;
+
+    // This will decide when battery levels change colors
+    // The values are in days left
+    private $greenBatteryLevelThreshold = 20; // More than 20 days = Green
+    private $yellowBatteryLevelThreshold = 5; // More than 5 days = Yellow
+    private $orangeBatteryLevelThreshold = 3; // More than 3 days = Orange
+                                              // Less than 3 days = Red
+
+    /**
+     * Device constructor.
+     * @param \Google\Cloud\Iot\V1\Device $device
+     */
+    public function __construct(\Google\Cloud\Iot\V1\Device $device)
+    {
+        parent::__construct($device);
+        if ($device->getLastEventTime() !== null && $device->getLastEventTime()->getSeconds() > 0) {
+            $this->lastMeasurement = date("Y-m-d  H:i:s", $device->getLastEventTime()->getSeconds());
+        } else {
+            $this->lastMeasurement = "Never";
+        }
+
+        $this->mockBattery(); // TODO: This should be replaced with the actual battery
+    }
+
+
+    /**
+     * Generates mock values for the battery levels
+     */
+    private function mockBattery() {
+        $this->batteryLevel = rand(0, 100);
+        $this->batterySecondsLeft = $this->batteryLevel * rand(10000,100000);
+    }
+
+    /**
+     * Generates the Battery text, and the progress bar showing how much battery is left
+     * @return string
+     * @throws Exception
+     */
+    public function getBatteryLevel()
+    {
+        if ($this->getEstimatedBatteryDaysLeft() > $this->greenBatteryLevelThreshold) {
+            $level = "green";
+            $progressColor = "bg-success";
+        } elseif ($this->getEstimatedBatteryDaysLeft() > $this->yellowBatteryLevelThreshold) {
+            $level = "yellow";
+            $progressColor = "bg-warning";
+        } elseif ($this->getEstimatedBatteryDaysLeft() > $this->orangeBatteryLevelThreshold) {
+            $level = "orange";
+            $progressColor = "bg-danger";
+
+        } else {
+            $level = "red";
+            $progressColor = "bg-danger";
+        }
+
+        $battery_text = "<span class='battery-level $level'>" . str_pad("$this->batteryLevel%", 10, " ", STR_PAD_RIGHT) . "</span> - " . $this->getEstimatedBatteryLeft($level);
+        $progressbar = "
+                        <div class=\"progress\">
+                            <div class=\"progress-bar $progressColor\" style=\"width: $this->batteryLevel%\" role=\"progressbar\" aria-valuenow=\"$this->batteryLevel\" aria-valuemin=\"0\" aria-valuemax=\"100\">
+                            </div>
+                        </div>";
+        return $battery_text . $progressbar;
+    }
+
+    /**
+     *  Returns the texts showing how many day(s) and hour(s) are left
+     * @param $class - CSS class of which color the text should be
+     * @return string
+     * @throws Exception
+     */
+    public function getEstimatedBatteryLeft($class)
```

Comment:
Should this and some of the following functions not be private?
```suggestion
    private function getEstimatedBatteryLeft($class)
```

Reply:
Does the fact that you resolved this mean that they should all be public?

Reply:
This function does no longer exist